subprojects/core/src/integTest/groovy/org/gradle/api/tasks/NestedInputIntegrationTest.groovy

```diff
@@ -820,7 +823,81 @@ class NestedInputIntegrationTest extends AbstractIntegrationSpec implements Dire
 
         then:
         executedAndNotSkipped(":customTask")
-        file("output.txt").text == "[100:example][100:example]"
+        file("output.txt").text ==~ /\[100:NestedBean@.*\]\[100:NestedBean@.*\]/
```

Comment:
🤔 Instead of trying to match the output of the default `Object.toString()` with a regexp, you could implement `NestedBean.toString()`, perhaps to print the value itself. That way you could still use regular string equality here.